string object
string color
int16 node
